File: desktop/src-tauri/src/commands/os_idle.rs
/// Seconds since the last OS-wide user input (keyboard/mouse anywhere on the
/// machine), or `None` on platforms without a supported idle API (Linux
/// Wayland). Callers fall back to in-app activity tracking when `None`.
#[tauri::command]
pub fn get_os_idle_seconds() -> Option<u64> {
    #[cfg(any(target_os = "macos", windows))]
    {
        user_idle::UserIdle::get_time()
            .ok()
            .map(|idle| idle.as_seconds())
    }
    #[cfg(not(any(target_os = "macos", windows)))]
    {
        None
    }
}

File: desktop/src/shared/api/osIdle.ts
import { invokeTauri } from "@/shared/api/tauri";

// Seconds since the last OS-wide user input, or null where the platform has
// no supported idle API (e.g. Linux Wayland).
export function getOsIdleSeconds(): Promise<number | null> {
  return invokeTauri<number | null>("get_os_idle_seconds");
}
